Payment methods: tweak border styles to remove radius and border on inactive/disabled items

ramonjd opened the pull request:
(Maybe) fixes https://github.com/Automattic/wp-calypso/issues/100156


## Proposed Changes


This PR update the payment method radio button and payment method styling to:

- Add hover and disabled state styling for radio buttons
- Adjust border radius for checked radio buttons
- Prevent border on disabled payment method hover state

### Before

https://github.com/user-attachments/assets/e67e8090-17f8-4e19-8808-fc32ca6fc82d



### After

https://github.com/user-attachments/assets/98338580-ee99-4c2e-9319-c4fb40cc0fe4



## Why are these changes being made?

- To remove the border-radii on inactive list items
- To remove the border on disabled list items

## Testing Instructions

Head over to `/me/purchases/[YOUR_SITE.wordpress.com]` and click on a product.

You'll be taken to `/me/purchases/[YOUR_SITE.wordpress.com]/[PRODUCT_ID]`

Hover over the payment items to ensure there are no regressions. They should have blue borders.

Unchecked items will not have the border radius at the edges.

Update a setting or click "Use this card" if you have one registered. Ensure that all items do not receive full borders. The unchecked items should retain their bottoms. Ho!



## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

ramonjd committed:
Improve radio button and payment method styling

- Add hover and disabled state styling for radio buttons
- Adjust border radius for checked radio buttons
- Prevent border on disabled payment method hover state

ramonjd committed:
Tightens up border states for disabled lists. Probably not really worth it, but makes hovering nicer while the form is disabled.

ramonjd commented on packages/composite-checkout/src/components/checkout-payment-methods.tsx:
```diff
@@ -23,6 +23,10 @@ const debug = debugFactory( 'composite-checkout:checkout-payment-methods' );
+	> div > div[disabled]:has( + div[disabled]:hover )::before,
+	> div > div[disabled]:has( + div.is-checked[disabled] )::before {
+		border-bottom: none;
+	}
```
All this faffery is to ensure hover states look okay when the form is disabled. 🤷🏻 

Not really necessary, but it's a detail.

matt-west commented:
Thanks @ramonjd. I noticed a couple things while testing this.

Let’s retain the border radius for the hover state. At the moment, we have sharp corners.

<img width="782" alt="Screenshot 2025-03-12 at 09 59 36" src="https://github.com/user-attachments/assets/2f70923e-8ddf-484a-aa36-dce1f2dc99b0" />

When the form is saving, all the options are disabled, but I still get a hover state for the options. We shouldn’t display the hover state if the option is disabled.

<img width="1135" alt="Screenshot 2025-03-12 at 09 55 45" src="https://github.com/user-attachments/assets/437731a7-8292-4952-bf23-48a523cb4a23" />

ramonjd commented:
> We shouldn’t display the hover state if the option is disabled.

Ah great, thanks for the feedback @matt-west - I think that simplify the CSS too.

ramonjd committed:
Enhance radio button and payment method styles

- Update hover state for enabled payment methods
- Refine border styles for disabled radio buttons
- Adjust border radius on hover for improved visual consistency

ramonjd commented:
Latest:

https://github.com/user-attachments/assets/c1a910f4-1914-489c-bf2c-fe3daf611ac2


I also tried to remove the sibling's bottom margin on hover or when selected as it was lurking around like a ghost artifact. 


Disabled states for selected items:


<img width="769" alt="Screenshot 2025-03-13 at 3 36 50 pm" src="https://github.com/user-attachments/assets/aa759be0-a29b-4127-ad90-9968222cd71b" />
<img width="737" alt="Screenshot 2025-03-13 at 3 37 03 pm" src="https://github.com/user-attachments/assets/51a915a6-12fc-4209-b118-98fc99e81b74" />

matt-west commented:
Thanks @ramonjd. Looks good now.

sirbrillig commented on packages/composite-checkout/src/components/radio-button.tsx:
```diff
@@ -1,4 +1,5 @@
 import styled from '@emotion/styled';
+import clsx from 'clsx';
```
The `composite-checkout` package does not have `clsx` as a dependency. This might work due to the nature of the monorepo but we shouldn't rely on it as it could break at any time. Do you need `clsx()` here? Or could you do this manually?

ramonjd replied:
Good catch. Done.

ramonjd committed:
Update radio-button.tsx

ramonjd committed:
Use the template literal approach instead, which is more explicit and will always return a string

sirbrillig reviewed (approved):
Thanks!
